include/circt/Dialect/HW/HWAggregates.td

```diff
@@ -139,6 +139,23 @@ def ArrayGetOp : HWOp<"array_get",
   let hasFolder = 1;
 }
 
+def ArrayInjectOp : HWOp<"array_inject",
+    [NoSideEffect,ArrayElementTypeConstraint<"newValue", "input">,
+     TypesMatchWith<"input and result types should be equivalent",
+                     "input", "result", [{ $_self }]>]> {
+  let summary = "Inject a value at the given index in an array";
+
+  let arguments = (ins ArrayType:$input, I32Attr:$index,
```

Comment:
I hate to ask, but is i32 big enough?

Reply:
Do you think it would be possible or sensible to use a dynamic Value for the index (i.e. HWIntegerType)? That would allow it to be more than just a compile time constant. I guess that would be trickier to emit as verilog... I guess that can be a future thing if it ever comes up.

Reply:
A plain IntegerAttr I thing should work for arbitrary sizes.  i32 is not enough.